src/utils.rs

```diff
@@ -282,6 +282,47 @@ pub fn calculate_sn_keccak(data: &[u8]) -> [u8; 32] {
     result
 }
 
+// ------------------------------------
+//  Invoke internal functions utils
+// ------------------------------------
+
+// Performs validation on fields related to function invocation transaction.
+// InvokeFunction transaction.
+// Deduces and returns fields required for hash calculation of
+
+pub(crate) fn preprocess_invoke_function_fields(
```

Comment:
Why this is here? we can move it to the module that contains the InvokeFunction struct since it is only useful there.

Reply:
fixed